### PR TITLE
task-driver: settle-match-external: Record historical fill and metrics

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR updates the `settle-external-match` task to record order fill metrics when an atomic match is witnessed on-chain. Note that if a client fails to submit the atomic match bundle on-chain the fill will be missed. For now, we choose not to solve this problem under the assumption that this is a sufficiently unlikely edge case. We can reevaluate if this makes sense once the atomic match system is deployed.

### Testing
- Unit tests pass
- Submitted an atomic match and printed the fill history after the match

### CODE REVIEW

File: workers/handshake-manager/src/manager/matching/external_engine.rs [hunk 1]
```diff
@@ -65,7 +66,12 @@ impl HandshakeExecutor {
             // may match the external order as the first party
             match_res.direction = order.side.opposite().match_direction();
             let settle_res = self
-                .try_settle_external_match(other_order_id, price, match_res, response_topic.clone())
```
Comment (akirillo) [inline workers/handshake-manager/src/manager/matching/external_engine.rs:68]: tiny nit: we can now use `ts_price.as_fixed_point()` when defining `price` above

**Review by akirillo — APPROVED** [review]
